### PR TITLE
Next dark theme

### PR DESCRIPTION
**Pull request by atanasster**
#### What does this PR do?
Fix anchor color on brand color and icons on brand color

#### What are the relevant issues?
#1982 
#1984 

#### Do the grommet docs need to be updated?
no
#### Is this change backwards compatible or is it a breaking change?
compatible

### CODE REVIEW

File: src/js/components/Select/README.md [hunk 1]
```diff
@@ -93,6 +93,14 @@ Whether when 'plain' it should receive a focus outline.
 boolean
 ```
 
+**label**
```
Comment (alansouzati) [inline src/js/components/Select/README.md]: why not using value ? I'm not sure why we need a separate prop for the label. in fact this seems not to be the label but the selected item in the select. value today can be a react node right ? so in theory you can pass anything to the selected item.

File: src/js/components/Select/SelectContainer.js [hunk 2]
```diff
@@ -60,7 +60,13 @@ class SelectContainer extends Component {
         nextValue = [];
         nextSelected = [];
         let removed = false;
-        (selected || []).forEach((selectedIndex) => {
+        let alreadySelected = [];
```
Comment (alansouzati) [inline src/js/components/Select/SelectContainer.js]: what is this doing ?

File: src/js/themes/dark.js [hunk 3]
```diff
@@ -18,7 +18,7 @@ const lightColors = ['#333333', '#444444', '#555555', '#666666', '#777777', '#99
 const darkColors = ['#F6F6F6', '#EEEEEE', '#DDDDDD', '#CCCCCC', '#BBBBBB', '#AAAAAA'];
 const backgroundColor = '#111111';
 const textColor = '#eeeeee';
-const borderColor = 'rgba(255, 255, 255, 0.33)';
+const borderColor = 'rgba(255, 255, 255, 0.50)';
```
Comment (alansouzati) [inline src/js/themes/dark.js]: why was this changed?

File: src/js/themes/dark.js [hunk 4]
```diff
@@ -66,7 +66,25 @@ export default deepFreeze({
...
+          props.theme.global.colors.lightBackground.text;
+      }
+      return `
+        fill: ${color};
+        stroke: ${color};
+      `;
+    }
+} `,
+  },
+  anchor: {
```
Comment (alansouzati) [inline src/js/themes/dark.js]: why was this added?

File: src/js/themes/dark.js [hunk 5]
```diff
@@ -66,7 +66,25 @@ export default deepFreeze({
...
   },
   icon: {
-    color: textColor,
+    extend: css`${(props) => {
+      let { color } = props.theme.icon;
+      if (props.color && props.color !== 'plain') {
+        color = colorForName(props.color, props.theme);
+      }
+      if (props.dark === colorIsDark(color)) {
+        color = props.dark ?
```
Comment (alansouzati) [inline src/js/themes/dark.js]: why do you need to check for `props.dark` here?

File: src/js/themes/dark.js (hunk 3 above)
Comment (atanasster) [reply src/js/themes/dark.js]: The brand color is pretty light, so the border is barely visible:
screenshot701

File: src/js/themes/dark.js (hunk 4 above)
Comment (atanasster) [reply src/js/themes/dark.js]: Not sure if you mean the anchor settings, but dark color Anchors are invisible when on brand background

File: src/js/themes/dark.js (hunk 5 above)
Comment (atanasster) [reply src/js/themes/dark.js]: is the user specified a color that would not be visible - for example a dark color on dark background, or light color on light background

**Comment by atanasster** [comment]
The PR of the select and dark theme have been added together, I am not really that good at separating PRs.

File: src/js/components/Select/SelectContainer.js (hunk 2 above)
Comment (atanasster) [reply src/js/components/Select/SelectContainer.js]: This is for #2007

File: src/js/themes/dark.js (hunk 4 above)
Comment (atanasster) [reply src/js/themes/dark.js]: yes, I do:
default theme (watch the anchor and select colors): https://grommet-nextjs.herokuapp.com/?theme=grommet 
dark theme (watch the anchor and select colors): https://grommet-nextjs.herokuapp.com/?theme=dark

**Comment by atanasster** [comment]
removed Select PR changes

File: src/js/themes/dark.js [hunk 6]
```diff
@@ -66,7 +66,25 @@ export default deepFreeze({
...
+      }
+      return `
+        fill: ${color};
+        stroke: ${color};
+      `;
+    }
+} `,
+  },
+  anchor: {
+    color: '#2b4369',
```
Comment (ericsoderberghp) [inline src/js/themes/dark.js]: This should be a formal color reference from `global.colors`.

Comment (atanasster) [reply src/js/themes/dark.js]: it was - inheriting from vanilla.js:
```
anchor: {
    textDecoration: 'none',
    fontWeight: 600,
    color: css`${props => props.theme.global.colors.brand}`,
  },
```
and we have invisible anchors on brand background:
screenshot689

https://grommet-nextjs.herokuapp.com/add-ons/numberinput?theme=dark

Comment (atanasster) [reply src/js/themes/dark.js]: Do as you please - close, reject or whatever, I am not touching this PR and not submitting anything else.
The theming in v2 based on dark/light background is a hack anyway but invisible Anchors on the brand menu seems like a major bug to me.

**Comment by alansouzati** [comment]
this still makes sense

**Comment by ericsoderberghp** [comment]
Is this pull request still needed, given https://github.com/grommet/grommet/pull/2151 ?

**Comment by atanasster** [comment]
#2151 does not fix the issue, actually makes it happen for grommet theme as well. Pr is not needed i guess but a fix is needed

**Comment by atanasster** [comment]
closing this PR, issues are still not fixed but the themes structure was changed since then.
